cpp/src/neighbors/specializations/ball_cover_knn_query.cu

```diff
@@ -14,18 +14,10 @@
  * limitations under the License.
  */
 
+#include <cstdint>
 #include <raft/neighbors/ball_cover.cuh>
 #include <raft/neighbors/ball_cover_types.hpp>
-
-// Ignore upstream specializations to avoid unnecessary recompiling
```

Comment:
Can you highlight what the issue here was and why it's not there anymore?

Reply:
The libraft-nn package and libraft-distance were independent and didn't depend on each other. This was done by design so they could be installed individually. As a result, we needed to explicitly check if the distance lib was also compiled/installed when the nn lib was being compiled so we could benefit from skipping those specializations. Otherwise we had to re-compile the stuff individually for each library.

This is why I mentioned last week (offline) that consolidating these libraries is going to lower the size and compile faster- all of the specializations can always benefit from all of the other specializations (eg disrance specializations get compiled once and can always be used by brute force knn, fused l2 knn, rbc now that they are all together in the same package).